vm/src/pyobject.rs

```diff
@@ -452,9 +452,24 @@ impl PyContext {
         )
     }
 
-    pub fn new_rustfunc(&self, function: RustPyFunc) -> PyObjectRef {
+    pub fn new_rustfunc<F: 'static + Fn(&mut VirtualMachine, PyFuncArgs) -> PyResult>(
+        &self,
+        function: F,
+    ) -> PyObjectRef {
         PyObject::new(
-            PyObjectKind::RustFunction { function: function },
+            PyObjectKind::RustFunction {
+                function: Box::new(function),
+            },
+            self.function_type(),
+        )
+    }
+
+    pub fn new_rustfunc_from_box(
+        &self,
+        function: Box<(Fn(&mut VirtualMachine, PyFuncArgs) -> PyResult)>,
```

Comment:
Can you use here `Box<RustPyFunc>`?

Reply:
Should I make a RustPyFunc trait? For example https://stackoverflow.com/a/26071172/9236675. I would have made it with just a type alias if it were possible, but the solution in the SO question seems kinda weird.

Reply:
Alright, I did that, cause it seems much cleaner than a bunch of `Fn(..) -> PyResult` everywhere.